src/ripple/net/SSLHTTPDownloader.h

```diff
@@ -41,7 +41,7 @@ namespace ripple {
 
 /** Provides an asynchronous HTTPS file downloader
  */
-class SSLHTTPDownloader
+class SSLHTTPDownloader : public std::enable_shared_from_this<SSLHTTPDownloader>
```

Comment:
Consider making the constructor private and adding a friend function:
```
    friend std::shared_ptr<DatabaseDownloader>
    make_DatabaseDownloader(
        boost::asio::io_service& io_service,
        Config const& config,
        beast::Journal j);
```
Rational: this class *must* be accessed through a shared pointer, so it would be good if the interface only allowed creating shared pointers. One downside: we can't use `make_shared` since it doesn't have access to the private constructor. This will cost us an extra allocation. However, given how infrequently this class is created that's probably an OK tradeoff.

Reply:
Cherry picked this change into my branch.